### PR TITLE
SONARJAVA-1170 remove AstNode

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/parser/sslr/SyntaxTreeCreator.java [hunk 1]
```diff
@@ -185,18 +149,28 @@ private InternalSyntaxToken visitTerminal(ParseNode node) {
       updateTokenPositionAndValue(node);
       TokenExpression ruleMatcher = (TokenExpression) node.getMatcher();
       tokenBuilder.setType(ruleMatcher.getTokenType());
+      isEof = ruleMatcher.getTokenType() == GenericTokenType.EOF;
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/parser/sslr/SyntaxTreeCreator.java]: MAJOR Change this comparison to use the equals method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)

File: java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java [hunk 2]
```diff
@@ -17,28 +17,50 @@
...
+    return true;
   }
 
+  @Override
+  public Iterator<Tree> childrenIterator() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void accept(TreeVisitor visitor) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java]: MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)

File: java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java [hunk 3]
```diff
@@ -17,28 +17,50 @@
...
+    this.start = start;
+    this.end = end;
+  }
 
   @Override
-  public void visitFile(AstNode astNode) {
-    File file = getContext().getFile();
-    getContext().addSourceCode(new SourceFile(file.getAbsolutePath(), file.getPath()));
+  public Kind getKind() {
+    // FIXME should have a dedicated kind associated with a dedicated interface.
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: java-squid/src/main/java/org/sonar/java/model/InternalSyntaxToken.java [hunk 4]
```diff
@@ -19,42 +19,54 @@
...
   private List<SyntaxTrivia> trivias;
-
+  private int startIndex;
+  private int endIndex;
+  private final int line;
+  private final int column;
+  private final String value;
+  private final boolean isEOF;
 
   protected InternalSyntaxToken(InternalSyntaxToken internalSyntaxToken) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/InternalSyntaxToken.java]: MAJOR Protected method '<init>(...)' is never used. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java [hunk 5]
```diff
@@ -116,14 +117,14 @@ public void visitFile(@Nullable AstNode astNode) {
...
-      JavaFileScannerContext context = new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors);
-      for (JavaFileScanner scanner : scanners) {
-        scanner.scanFile(context);
-      }
-      if (semanticModel != null) {
-        // Close class loader after all the checks.
-        semanticModel.done();
-      }
+    }
+    JavaFileScannerContext context = new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors);
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: MAJOR Rename "context" which hides the field declared at line 68. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 6 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this unused "name" private field.](https://github.com/SonarSource/sonar-java/blob/faa1796afe15c66fac1cd6d60c841ba2669444d1/java-squid/src/main/java/org/sonar/java/model/JavaTree.java#L420) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  public InternalSyntaxSpacing(int start, int end) {
+    super(null);
+    this.start = start;
+    this.end = end;
+  }
+
+  @Override
+  public Kind getKind() {
+    // FIXME should have a dedicated kind associated with a dedicated interface.
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java [hunk 7]
```diff
@@ -116,14 +117,14 @@ public void visitFile(@Nullable AstNode astNode) {
...
-      JavaFileScannerContext context = new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors);
-      for (JavaFileScanner scanner : scanners) {
-        scanner.scanFile(context);
-      }
-      if (semanticModel != null) {
-        // Close class loader after all the checks.
-        semanticModel.done();
-      }
+    }
+    JavaFileScannerContext javaFileScannerContext = new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors);
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: MINOR Split this 193 characters long line (which is greater than 180 authorized). [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 3 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by m-g-sonar** [comment]
The coupling between `VisitorBridge` and `JavaAstScanner` could probably be slightly simplified, but LGTM! Let's eradicate AstNode!

**Pull request merged**